File: core/src/main/java/eu/kotkas/villak/core/villak/model/Game.java
package eu.kotkas.villak.core.villak.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

/**
 * @author Kristen Kotkas
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Game implements Serializable {

  private List<Team> teams;
  private List<Round> rounds;
  @Builder.Default
  private Settings settings = new Settings();

}

File: web/src/app/modules/villak/game-data/model/game.ts
import {Round} from './round';
import {Team} from './team';
import {Settings} from "./settings";

export class Game {
  teams: Team[] = undefined;
  rounds: Round[] = undefined;
  settings: Settings = undefined;

  public static parseGame(input: any): Game {
    const json = JSON.parse(input);
    return {
      teams: json['teams'],
      rounds: json['rounds'],
      settings: json['settings']
    };
  }

  constructor(teams: Team[], rounds: Round[]) {
    this.teams = teams;
    this.rounds = rounds;
  }
}
